### DISCUSSION TITLE
Nuxt 3 custom composable

### DISCUSSION THREAD

**Discussion by monkemedia** [nuxt/nuxt#16509] [Questions]
Wondering if anyone can help me. This is how I would usually write a custom composable

Template (app.vue):
```
const { load, result } = useCart()

useAsyncData('cart', () => load())

console.log('result', result.value)
```

Composable (useCart()):
```
const result: any = ref([])

export function useCart () {
  const fetchData = fetchApi()
  const loading = ref(false)
  const error = ref(null)
  
  const load = async () => {
    loading.value = true

    try {
      const { data } = await fetchData.get(`/carts/123`)
      result.value = data.data
    } catch (err) {
      error.value = err
    } finally {
      loading.value = false
    }
  }

  return {
    load,
    result,
    loading,
    error
  }
}
```

This would usually serve me well, especially with something like a cart function, where I would want to get access to the cart (results) property anywhere in my application. In this case, the cart icon updating in my navbar, while the user updates the cart on the product page for example.

Since upgrading to NUXT 3 rc-12, serverside data is no longer being passed to the clientside using the method above. The only way that seems to resolve this issue is by returning the data like this:

```
const load = async () => {
      const { data } = await fetchData.get(`/carts/123`)
      return data.data
 }
```

and then calling it like this in my template:

```
const { data } = useAsyncData('cart', () => load())
console.log('data', data)
```

This corrects the issue with the data being served correctly on the clientside, however, I no longer have access to the data globally. ie my cart icon in the navbar does not have access to the data unless I call the load method again. This would equal poor performance making the request again :(

**Comment by wobsoriano** [comment]
Are you using [useFetch](https://v3.nuxtjs.org/api/composables/use-fetch/) in that `fetchApi()` thing?

**Reply by monkemedia** [reply]
Im using Axios at the moment as I have been using its interceptors. But have been contemplating on changing it to useFetch or $fetch
